tls/s2n_renegotiate.c

```diff
@@ -20,6 +20,26 @@
 #include "tls/s2n_connection.h"
 #include "utils/s2n_safety.h"
 
+/* We don't want to introduce a new blocked status for renegotiation
+ * because that would potentially require applications to update their
+ * blocked status handling logic for no reason.
+ *
+ * It is impossible to use both early data and renegotiation for the same handshake,
+ * and both are just application data received during the handshake.
+ * Therefore, we will alias S2N_BLOCKED_ON_EARLY_DATA for reuse with renegotiation.
+ */
+const s2n_blocked_status S2N_BLOCKED_ON_APPLICATION_DATA = S2N_BLOCKED_ON_EARLY_DATA;
+
+S2N_RESULT s2n_renegotiate_validate(struct s2n_connection *conn)
+{
+    RESULT_ENSURE_REF(conn);
+    RESULT_ENSURE(s2n_in_unit_test(), S2N_ERR_NOT_IN_UNIT_TEST);
```

Comment:
Why would we only allow this in unit tests?

Reply:
That's just to keep it gated to the unit tests for now. I'll remove that line when this becomes publicly accessible.